qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -238,20 +260,45 @@ def jobs(
         if instance:
             api_filter["provider"] = instance
         # Retrieve all requested jobs.
-        job_responses = self._get_jobs(
-            api_filter=api_filter, limit=limit, skip=skip, descending=descending
+        filter_by_status = (
+            status
+            and status not in ["pending", "completed"]
+            and status in [status.name for status in JobStatus]
         )
         job_list = []
-        for job_info in job_responses:
-            # TODO filter by status
-            job = self._restore_circuit_job(job_info, raise_error=False)
-            if job is None:
-                logger.warning(
-                    'Discarding job "%s" because it contains invalid data.',
-                    job_info.get("job_id", ""),
-                )
-                continue
-            job_list.append(job)
+        original_limit = limit
+        while True:
+            job_responses = self._get_jobs(
+                api_filter=api_filter, limit=limit, skip=skip, descending=descending
+            )
+            if len(job_responses) == 0:
+                break
+            different_status_jobs = 0
+            for job_info in job_responses:
+                if filter_by_status:
+                    job_status = api_status_to_job_status(
+                        job_info["state"]["status"].upper()
+                    ).name
+                    if (isinstance(status, str) and job_status != status) or (
+                        isinstance(status, list) and job_status not in status
+                    ):
+                        continue
+                    different_status_jobs += 1
+                job = self._restore_circuit_job(job_info, raise_error=False)
+                if job is None:
+                    logger.warning(
+                        'Discarding job "%s" because it contains invalid data.',
+                        job_info.get("job_id", ""),
+                    )
+                    continue
+                job_list.append(job)
+                if (
+                    len(job_list) == original_limit
+                    or different_status_jobs == original_limit
```

Comment:
Why do you need to check `different_status_jobs` here?

Reply:
I overcomplicated the logic here, `different_status_jobs` is not necessary